### PR TITLE
Factor out a function to load app config

### PR DESCRIPTION
**Pull request by chadwhitacre**
Part of #482.

Since we're going to have somewhat complex logic to parse envvars into app config, we should test as we go. Factoring out a function gives us something to test.

### CODE REVIEW

File: src/api/github/appAuthStrategyOptions.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// I didn't find something great in @octokit/types.
+
+export interface AppAuthStrategyOptions {
```
Comment (hubertdeng123) [inline src/api/github/appAuthStrategyOptions.ts:3]: should this belong in `src/types` instead?

File: src/config/loadGitHubApps.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { AppAuthStrategyOptions } from '@api/github/appAuthStrategyOptions';
+
+export function loadGitHubApps(env) {
```
Comment (hubertdeng123) [inline src/config/loadGitHubApps.ts:3]: does this function need to pass in environment variables as `env`? Can we just use `process.env` here?

**Review by hubertdeng123 — CHANGES_REQUESTED** [review]

File: src/config/loadGitHubApps.ts (hunk 2 above)
Comment (chadwhitacre) [reply src/config/loadGitHubApps.ts:3]: It makes testing much easier.

File: src/api/github/appAuthStrategyOptions.ts (hunk 1 above)
Comment (chadwhitacre) [reply src/api/github/appAuthStrategyOptions.ts]: Oooh, good call! Done in b98db3a7bea4da859f66107157f227636c120bac.

File: src/config/loadGitHubApps.ts (hunk 2 above)
Comment (hubertdeng123) [reply src/config/loadGitHubApps.ts:3]: In that case, can we just pass in `GH_APP_IDENTIFIER` and `GH_APP_SECRET_KEY` instead? feels weird to me to pass all the env variables just to use two of them

Comment (chadwhitacre) [reply src/config/loadGitHubApps.ts]: Filtering out unused envvars is part of the logic we want to test, no? Does da690d21b0e35db187ab60f147291a4d9e40a834 help? It changes the function name to `loadGitHubAppsFromEnvironment` and adds a few more tests.

Comment (chadwhitacre) [reply src/config/loadGitHubApps.ts]: Also keep in mind that the next PR is going to add multi-org support, for which we're going to need to iterate over the whole process.env to pull out the vars we want.

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**